Setting `exactOptionalPropertyTypes` results in `TS2720` errors in `vscode-languageserver-protocol`

With [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) turned on in my `tsconfig.json`, I get these errors from `vscode-languageserver-protocol`:

```
node_modules/vscode-languageserver-protocol/lib/common/messages.d.ts:15:22 - error TS2720: Class 'ProtocolRequestType0<R, PR, E, RO>' incorrectly implements class 'ProgressType<PR>'. Did you mean to extend 'ProgressType<PR>' and inherit its members as a subclass?
  Types of property '_pr' are incompatible.
    Type 'PR | undefined' is not assignable to type 'PR'.
      'PR' could be instantiated with an arbitrary type which could be unrelated to 'PR | undefined'.

15 export declare class ProtocolRequestType0<R, PR, E, RO> extends RequestType0<R, E> implements ProgressType<PR>, RegistrationType<RO> {
                        ~~~~~~~~~~~~~~~~~~~~

node_modules/vscode-languageserver-protocol/lib/common/messages.d.ts:25:22 - error TS2720: Class 'ProtocolRequestType<P, R, PR, E, RO>' incorrectly implements class 'ProgressType<PR>'. Did you mean to extend 'ProgressType<PR>' and inherit its members as a subclass?
  Types of property '_pr' are incompatible.
    Type 'PR | undefined' is not assignable to type 'PR'.
      'PR' could be instantiated with an arbitrary type which could be unrelated to 'PR | undefined'.

25 export declare class ProtocolRequestType<P, R, PR, E, RO> extends RequestType<P, R, E> implements ProgressType<PR>, RegistrationType<RO> {
                        ~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/vscode-languageserver-protocol/lib/common/messages.d.ts
```

I'm not familiar enough with the code to know if these represent actual problems, but I thought I'd open an issue just in case.

--- a/jsonrpc/src/common/connection.ts
+++ b/jsonrpc/src/common/connection.ts
@@ -59,8 +59,8 @@ export class ProgressType<PR> {
 	 * Clients must not use these properties. They are here to ensure correct typing.
 	 * in TypeScript
 	 */
-	public readonly __?: [PR, _EM];
-	public readonly _pr?: PR;
+	public readonly __: [PR, _EM] | undefined;
+	public readonly _pr: PR | undefined;
 
 	constructor() {
 	}

--- a/protocol/src/common/messages.ts
+++ b/protocol/src/common/messages.ts
@@ -28,6 +28,7 @@ export class ProtocolRequestType0<R, PR, E, RO> extends RequestType0<R, E> imple
 	 * Clients must not use these properties. They are here to ensure correct typing.
 	 * in TypeScript
 	 */
+	public readonly __: [PR, _EM] | undefined;
 	public readonly ___: [PR, RO, _EM] | undefined;
 	public readonly ____: [RO, _EM] | undefined;
 	public readonly _pr: PR | undefined;
@@ -41,6 +42,7 @@ export class ProtocolRequestType<P, R, PR, E, RO> extends RequestType<P, R, E> i
 	/**
 	 * Clients must not use this property. It is here to ensure correct typing.
 	 */
+	public readonly __: [PR, _EM] | undefined;
 	public readonly ___: [PR, RO, _EM] | undefined;
 	public readonly ____: [RO, _EM] | undefined;
 	public readonly _pr: PR | undefined;

